src/MadLibEntry/MadLibEntry.js

```diff
@@ -21,15 +21,22 @@ const MadLibEntry = ({ madLib }) => {
       str = str.replaceAll(key, userInputs[key])
     })
     console.log(str)
-    setResult(str)
+    setResult({result: str, isFavorited: false})
     updateDisplay()
   }
   
+  const favoriteResult = () => {
+    setResult({...result, isFavorited: true})
+  }
+
+  useEffect(() => {
+    result.isFavorited && addToFavorites(result)
+  }, [result.isFavorited])
```

Comment:
Curious about this. Is there a reason why `favoriteResult()` doesn't both `setResult(...)` and invoke `addToFavorites(result)`? Seems like you could lose the `useEffect` in that case, unless there's something else I'm not seeing yet.

Reply:
My understanding of it is since setting the state here is async, if I had `addToFavorites()` included right after `setResult` in `favoriteResult` it wouldn't register that the `isFavorited` property had been changed to true and wouldn't run the `addToFavorites()` function. So I have the UseEffect here basically to say, "once the `isFavorited` property is updated, _then_ run my `addToFavorites()` function. "
Truthfully, I'm a bit shaky on useEffect generally, so my understanding of how that works might not be based on truth!

Reply:
I see your point and I think you're right about setting state being async. I guess what I meant was that you could "disconnect" the expressions' dependencies on one another so that the user's click is essentially both changing local state (`result.isFavorited`) and `favorites` state with `addToFavorites()`. But the way you've done it—where the state of `favorites` is dependent on `result.isFavorited`—might actually be better practice, since they're less likely to get out of sync with each other/no way for the lib to be added to `favorites` state without local state reflecting that too. 

I'm actually not sure on this one. Thanks for explaining though! Something to think about.